File: src/main/java/fun/bookish/vertx/activemq/client/subscriber/ActiveMQSubscriber.java
package fun.bookish.vertx.activemq.client.subscriber;


import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;
import io.vertx.core.json.JsonObject;

public interface ActiveMQSubscriber {

    String getKey();

    void listen(Handler<AsyncResult<JsonObject>> messageHandler);

    void close();

    void close(Handler<AsyncResult<Void>> handler);

}

File: src/main/kotlin/cn/booklish/vertx/activemq/client/subscriber/ActiveMQSubscriber.kt
package cn.booklish.vertx.activemq.client.subscriber

import io.vertx.core.AsyncResult
import io.vertx.core.Handler
import io.vertx.core.json.JsonObject


interface ActiveMQSubscriber {

    fun listen(messageHandler: Handler<AsyncResult<JsonObject>>)

    fun close()

    fun close(handler: Handler<AsyncResult<Void>>)

}